File: php/test/BillingServiceTest.php
<?php


use PHPUnit\Framework\TestCase;

class BillingServiceTest extends TestCase
{
    /** @test */
    public function Name()
    {

        $oldBillingSystem = $this->createMock(OldBillingSystem::class);
        $billingService = new BillingService($oldBillingSystem);

        $oldBillingSystem->expects($this->once())
            ->method('invoice')
            ->with($this->equalTo("100$ to destination company"));

        $billingService->invoice(new Invoice(100, Currency::dollar(), "destination company"));


    }


}

File: csharp/test/BillingServiceTest.cs
using System;
using lib;
using lib.ExternalSDKs;
using Moq;
using Xunit;

namespace test
{
    public class BillingServiceTest
    {
        [Fact]
        public void TestBillingService()
        {

            var billingProviderMock = new Mock<GoodOldBillingProvider>();
            var service = new BillingService(billingProviderMock.Object);
            
            service.Invoice(new Invoice(100, Currency.Dollar(), "Vance Refrigeration"));
            
            billingProviderMock.Verify(provider => provider.SendInvoice("100$ to Vance Refrigeration"));
        }
    }

}
